src/com/namelessmc/java_api/NamelessUser.java

```diff
@@ -20,33 +12,47 @@
 import com.namelessmc.java_api.exception.ReportUserBannedException;
 import com.namelessmc.java_api.exception.UnableToCreateReportException;
 
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
 public final class NamelessUser {
 
 	private final NamelessAPI api;
 	private final RequestHandler requests;
-	
+
 	private Integer id;
 	private String username;
 	private Optional<UUID> uuid;
 	private Optional<Long> discordId;
 
 	private JsonObject userInfo;
-	
+
+	// only one of id, username, uuid, discordId has to be provided
+	@Deprecated
+	NamelessUser(final NamelessAPI api, final Integer id, final String username, final Optional<UUID> uuid, final Long discordId) {
+		this(api, id, username, uuid.orElse(null), discordId);
+	}
+
 	// only one of id, username, uuid, discordId has to be provided
-	NamelessUser(final NamelessAPI api, final Integer id, final String username, final Optional<UUID> uuid, final Long discordId) throws NamelessException {
+	NamelessUser(final NamelessAPI api, final int id, final String username, final UUID uuid, final long discordId) {
 		this.api = api;
 		this.requests = api.getRequestHandler();
-		
-		if (id == null && username == null && uuid == null && discordId == null) {
+
+		if (id == -1 && username == null && uuid == null && discordId == -1) {
 			throw new IllegalArgumentException("You must specify at least one of ID, uuid, username, discordId");
 		}
-		
+
 		this.id = id;
 		this.username = username;
-		this.uuid = uuid;
-		this.discordId = discordId != null ? Optional.of(discordId) : null;
+		this.uuid = Optional.ofNullable(uuid);
```

Comment:
I don't think this will work, keep in mind that there were three possible states:
- User has UUID (optional present)
- User does not have UUID (minecraft integration disabled or UUID not set on the website)
- UUID not retrieved yet, don't know if it exists or not (null)

Reply:
First Case: The UUID is in the Optional
Second Case: Empty Optional
Third Case: Empty Optional

Reply:
Also, an Optional should not be passed as null, as that completly destroys the reason for an Optional

Reply:
I know it goes against the standard optional usage pattern, I did it because it was the simplest way to represent the three states and this constructor is not public API anyway.

The differentiation between knowing there is no UUID and not knowing if the player has a UUID is important

Reply:
Ok, if that is needed, I will revert this change